src/api/entities/ApiRole.ts

```diff
@@ -0,0 +1,34 @@
+import { Model } from 'objection';
+import { z } from 'zod';
+
+import { BaseModel } from './BaseModel';
+import { ModelObjectOpt } from './ModelObjectOpt';
+
+export class ApiRole extends BaseModel {
+  static get tableName() {
+    return 'apiRoles';
+  }
+  static relationMappings = {
+    participants: {
+      relation: Model.ManyToManyRelation,
+      modelClass: 'Participant',
+      join: {
+        from: 'apiRoles.id',
+        through: {
+          from: 'participantsToRoles.apiRoleId',
+          to: 'participantsToRoles.participantId',
+        },
+        to: 'participants.id',
+      },
+    },
+  };
+  declare id: number;
+  declare roleName: string;
+}
+
+export type ApiRoleDTO = ModelObjectOpt<ApiRole>;
+
+export const ApiRoleTypeSchema = z.object({
```

Comment:
These schema are just for parsing - I know we have a few in models, but we've seen some bugs where people assume they're shared and should model the entity itself, and change them. I think they should really be closer to the thing that's parsing something (probably the route handlers).

Reply:
I think for now it may still be needed as it's used by the Participant Schema?